src/Statistics.h

```diff
@@ -133,30 +261,267 @@ class cStatManager
 
 	// tolua_begin
 
-	/** Return the value of the specified stat. */
+	/** Return the value of the specified stat as enum.
+		this function is still here to provide access for the old API */
 	StatValue GetValue(const eStatistic a_Stat) const;
 
-	/** Set the value of the specified stat. */
+	/** Return the value of the specified stat as enum class. */
+	StatValue GetValue(const Statistic a_Stat) const;
+
+	/** Set the value of the specified stat as enum.
+		this function is still here to provide access for the old API */
 	void SetValue(const eStatistic a_Stat, const StatValue a_Value);
 
+	/** Set the value of the specified stat as enum class. */
+	void SetValue(const Statistic a_Stat, const StatValue a_Value);
+
 	/** Reset everything. */
 	void Reset();
 
-	/** Increments the specified stat.
-	Returns the new value.
-	*/
+	/** Increments the specified stat as enum. Returns the new value.
+		this function is still here to provide access for the old API */
 	StatValue AddValue(const eStatistic a_Stat, const StatValue a_Delta = 1);
 
-	// tolua_end
+	/** Increments the specified stat as enum class. Returns the new value. */
+	StatValue AddValue(const Statistic a_Stat, const StatValue a_Delta = 1);
 
-private:
+	// tolua_end
+	/** Type -> Name */
+	static const AString GetName(const Statistic a_Type);
 
-	StatValue m_MainStats[statCount];
+	static const Statistic GetType(const AString & a_Type);
 
-	// TODO 10-05-2014 xdot: Use, mine, craft statistics
+	/** returns the prerequisite for a achievement */
+	static const Statistic GetPrerequisite(const Statistic a_Type);
 
+	/** returns the map of dirty statistics */
+	const std::map<Statistic, bool> * GetDirtyStatistics() const;
 
+private:
+	static const std::map<AString, Statistic> m_StatNames;
+
+	std::map<Statistic, int> m_MainStats=
```

Comment:
Could we not use the fixed size array (`int m_MainStats[Statistic::Count]`) that was there before?

Reply:
yes, we could do this - should improve the time

Reply:
thanks! also, if the count of stats isn't going to go negative, consider making it `unsigned`

Reply:
changed the dirty statistics too